stock_measuring_device/models/measuring_device.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+import logging
+
+from odoo import _, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class MeasuringDevice(models.Model):
+    _name = "measuring.device"
+    _inherit = "collection.base"
+    _description = "Measuring and Weighing Device"
+    _order = "warehouse_id, name"
+
+    name = fields.Char(required=True)
+    warehouse_id = fields.Many2one("stock.warehouse", required=True)
+    device_type = fields.Selection(
+        selection=[],
+        help="The type of device (e.g. zippcube, cubiscan...) "
+        "depending on which module are installed.",
+    )
+    state = fields.Selection(
+        [("not_ready", "Not Ready"), ("ready", "Ready")],
+        default="not_ready",
+        readonly=True,
+        copy=False,
+    )
+
+    _sql_constraints = [
+        (
+            "name_uniq",
+            "unique (name)",
+            "The name of the measuring/weighing device must be unique.",
+        ),
+    ]
+
+    def _get_measuring_device(self):
+        with self.work_on(self._name) as work_ctx:
+            return work_ctx.component(usage=self.device_type)
+
+    def open_wizard(self):
+        res = {
+            "name": _("Measurement Wizard"),
+            "res_model": "measuring.wizard",
+            "type": "ir.actions.act_window",
+            "view_id": False,
+            "view_mode": "form",
+            "context": {"default_device_id": self.id},
+            "target": "fullscreen",
+            "flags": {
+                "withControlPanel": False,
+                "form_view_initial_mode": "edit",
+                "no_breadcrumbs": True,
+            },
+        }
+        if self._is_being_used():
+            pack = self.env["product.packaging"].search(
+                [("measuring_device_id", "=", self.id)], limit=1
+            )
+            res["context"]["default_product_id"] = pack.product_id.id
+        return res
+
+    def _is_being_used(self):
+        self.ensure_one()
+        return bool(
+            self.env["product.packaging"].search_count(
+                [("measuring_device_id", "=", self.id)]
```

Comment:
```suggestion
                [("measuring_device_id", "=", self.id)], limit=1
```

nitpicking: preferred in this pattern

perhaps not so interesting here, as I don't think we can expect too many records anyway.. but existing code is future code's inspiration